micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jersey/server/MetricsRequestEventListenerTest.java

```diff
@@ -163,6 +169,11 @@ void exceptionsAreMappedCorrectly() {
             .tags(tagsFrom("/throws-mappable-exception", "410", "CLIENT_ERROR", "ResourceGoneException"))
             .timer()
             .count()).isEqualTo(1);
+
+        assertThat(registry.get(METRIC_NAME)
+            .tags(tagsFrom("root", "406", "CLIENT_ERROR", "NotAcceptableException"))
```

Comment:
The URI being `root` seems incorrect. Shouldn't we be able to capture `produces-text-plain` here?

Reply:
It seems that when the requested accepted media type is different than the media type supported by the endpoint, the `uri` is not matched by Jersey and by default is resolved to `root`.

We could probably change this behavior [here](https://github.com/micrometer-metrics/micrometer/blob/f5f3e013516172948e497769daa48496fc78224f/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jersey/server/JerseyTags.java#L97), but I'm not sure what kind of `uri` would be more desired in such case.

Reply:
I'm not seeing a good way we can get the URI template we want since the request was not matched to a resource. In that case, I think it would be best to use the value `UNKNOWN` for the `uri` tag rather than `root`. Using `root` makes it seem like the request was for a resource mapped to the root path, but it isn't necessarily and in the test case it definitely isn't.

Reply:
@adpaste would you be willing to update the PR to make sure `UNKNOWN` is used for the `uri` tag instead of `root` in this case?

Reply:
@shakuzen Sure, I've updated the PR with the proposed change. Since I'm not fully familiar with the codebase, could you please confirm if this approach looks alright to you?

Reply:
Thanks. Looks good to me.